extensions/wikia/PhalanxII/models/PhalanxContentModel.php

```diff
@@ -18,11 +18,13 @@ public function __construct( Title $title ) {
 	 *
 	 * @return bool
 	 */
-	public function isOk() {
+	public function isOk(): bool {
+		global $wgUser;
+
 		return (
```

Comment:
@mixth-sense 

Definitely! My rearranging of the conditions was with this in mind, although with the following reasoning:

* While we do enforce a strict type constraint on `$title` parameter in constructor, it is still theoretically possible (albeit highly unlikely) for it to receive `null` when it is called from hook, and this sanity check accounts for this possibility.
* And then there is the eternal struggle between `global $wgFoo` vs our Nirvana helpers `$this->wg->Foo`. The problem here is that `wg->Foo` is not actually direct property access (if it was, it would be no question it'd be better). `WikiaGlobalRegistry` does not actually store its elements as its own properties - instead, it provides magic `__get` method which then fetches the variable from `$GLOBALS` superglobal while performing sanity checks on the variable name. So using `global $wgFoo` is just creating a new reference, while `$this->wg->Foo` triggers 3 function calls before returning reference from `$GLOBALS`.

This is something I have been contemplating for a while, and I think it'd be actually beneficial in the long run to move away from Nirvana's `WikiaGlobalRegistry`. @macbre probably has more info on this, but I believe one of its main purposes was to ease testing and mocking globals before we had W(U)TF and test helpers. As it stands, the only useful feature it offers is synchronizing values with `RequestContext` main instance, which is of dubious value since most of the time we only read globals with this helper instead of overwriting them.

So in the spirit of #856 I think it is worth giving this a thought. MediaWiki 1.19 also introduced `RequestContext` which provides a way to move away from globals. Many of our common entry points where we hook into MW code already have a way to access local request context (e.g. through `OutputPage` or `Skin` instance), which would allow us to use this local context in our handlers and pass it along as required.